mkosi/qemu.py

```diff
@@ -359,21 +378,18 @@ def run_qemu(args: MkosiArgs, config: MkosiConfig, qemu_device_fds: Mapping[Qemu
     if (config.runtime_trees and config.qemu_firmware == QemuFirmware.bios):
         die("RuntimeTrees= cannot be used when booting in BIOS firmware")
 
-    if config.qemu_kvm == ConfigFeature.enabled and not QemuDeviceNode.kvm.available():
+    if config.qemu_kvm == ConfigFeature.enabled and not config.architecture.is_native():
+        die(f"KVM acceleration requested but {config.architecture} does not match the native host architecture")
+
+    have_kvm = (config.qemu_kvm != ConfigFeature.disabled and
+                (qemu_version(config) < QEMU_KVM_DEVICE_VERSION and QemuDeviceNode.kvm.available()) or
+                (qemu_version(config) >= QEMU_KVM_DEVICE_VERSION and QemuDeviceNode.kvm in qemu_device_fds))
+    if (config.qemu_kvm == ConfigFeature.enabled and not have_kvm):
```

Comment:
Checking `config.qemu_kvm` being enabled and being unequal from disabled seems a bit weird - it is tristate, but once it's enabled it can't be anything else anymore
```suggestion
    have_kvm = (
        (qemu_version(config) < QEMU_KVM_DEVICE_VERSION and QemuDeviceNode.kvm.available()) or
        (qemu_version(config) >= QEMU_KVM_DEVICE_VERSION and QemuDeviceNode.kvm in qemu_device_fds)
    )
    if (config.qemu_kvm == ConfigFeature.enabled and not have_kvm):
```

Reply:
But this means we'll try to open /dev/kvm even if the feature is disabled which does not seem desirable?

Reply:
Although this shouldn't really matter, let me rework it like you suggested